src/tuplet.ts

```diff
@@ -73,7 +73,7 @@ export class Tuplet extends Element {
   protected y_pos: number;
   protected x_pos: number;
   protected width: number;
-  protected location: number;
+  protected location!: number;
```

Comment:
The linter does not comply?

Reply:
It seems like adding a ! is equivalent to asking the TS compiler to avoid flagging this line. It's basically like saying "trust me, I know what I am doing here because I actually do initialize this property later".

eslint does not complain. Perhaps there is a setting we can turn on if we care to.

Reply:
More info here:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#definite-assignment-assertions

We definitely need to be careful when using the **definite assignment** operator. (It's similar to the non-null assertion, which the eslint will definitely complain about.)

For now, we can just require a comment above the instance variable that indicates where it is assigned. In this case, I will need to add a comment to tuplet.ts to show where `location` is assigned.

Reply:
It's actually easy to search for where we use this feature.

Doing a project wide search for the string `!:` I see that I have 40 instances of it. The master version has 34. So I will need to check where I added them and provide comments to verify that those instance variables are definitely assigned via a helper function.